parsing/ast_invariants.ml

```diff
@@ -45,6 +48,14 @@ let simple_longident id =
   in
   if not (is_simple id.txt) then complex_id id.loc
 
+let not_optional_label loc l =
+  let is_optional =
+    match l with
+    | Optional _ -> true
+    | _ -> false
+  in
+  if is_optional then optional_label_on_functor loc
```

Comment:
Note that you could test the invariant in `testsuite/tests/parsing/broken_invariants.ml`.